### PR TITLE
Migrating from GH OAuth to GH App

### PR DESCRIPTION
**Pull request by helinanever**
Added a migration guide

### CODE REVIEW

**Comment by helinanever** [comment]
Changes to ../getting-started/signup/#selected-repositories-github-app will be made in a separate PR.

File: content/getting-started/signup.md [hunk 1]
```diff
@@ -46,31 +46,33 @@ If you already have a Codemagic account, you can enable the GitHub app integrati
...
-* The primary team owner has connected the GitHub app integration for their personal account in **User settings > Integrations** and the team has not enabled the GitHub integration. If the regular GitHub integration is enabled for the team, the connected account makes use of OAuth permissions and the restrictions you configure in GitHub app do not apply.
+* GitHub App integration is connected in **Team settings > Team integrations**.
+* The primary team owner has connected the GitHub App integration for their personal account in **User settings > Integrations** and the team has not enabled the GitHub (OAuth) integration. If the regular GitHub integration is enabled for the team, the connected account makes use of OAuth permissions and the restrictions you configure in GitHub App do not apply.
 
 {{<notebox>}}
 Currently, Codemagic has no way of knowing whether it has access to team apps before the build is started. Due to this, team apps to which you have revoked access in the GitHub app may not appear grayed out in the UI.
 {{</notebox>}}
 
-#### Disabling Codemagic GitHub app integration
+#### Diconnecting Codemagic GitHub app integration
```
Comment (martlaub) [inline content/getting-started/signup.md:73]: Capitalize "app"

File: content/knowledge-base/migrate-to-github-app.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## GitHub OAuth vs GitHub App
+
+The key difference between OAuth and GitHub App integrations is the scope of permissions. While OAuth requires you to grant Codemagic **write access** to all your repositories, the GitHub App integration requests only **read access** to your code and you can select which repositories you share with Codemagic.
+
+In addition, the GitHub app integration will also make it possible to use [GitHub Checks](../building/github-checks).
+
+### Deprecting GitHub releases in Flutter workflow editor
+
+Without write access to the repository, Codemagic will no longer be able to push GitHub releases without additional configuration. Therefore, we will be removing the GitHub releases section in Flutter workflow editor. You can continue to
+publish GitHub releases by setting up a personal access token and using a custom script in the ... step.
```
Comment (martlaub) [inline content/knowledge-base/migrate-to-github-app.md:25]: TODO: Finish.

Comment (priitlatt) [inline content/knowledge-base/migrate-to-github-app.md]:
```suggestion
publish GitHub releases by setting up a personal access token and using a custom script in the Pre-publish step.

```bash
#!/usr/bin/env zsh

# Make sure that you have exported your GitHub personal access token as
# environment variable `GITHUB_TOKEN`. GitHub CLI uses token from this 
# environment variable for authentication.

# Publish only for tag builds
if [ -z ${FCI_TAG} ]; then
   echo "Not a tag build will not publish GitHub release"
   exit 0
fi

# See more options about `gh release create` usage from GitHub CLI
# official docs at https://cli.github.com/manual/gh_release_create

gh release create "${FCI_TAG}" \
    --title "<Your Application Name> ${FCI_TAG}" \
    --notes-file changelog.md
    path/to/build-artifact.ipa \
    path/to/build-artifact.apk
```

# Note that you don't need to include title and changelog if you do not want to.
# Any number of artifacts can be included with the release.

**Review by martlaub — APPROVED** [review]

**Pull request merged**